src/components/Tooltip.tsx

```diff
@@ -0,0 +1,275 @@
+ * );
+ *
+ * export default MyComponent;
+ * ```
+ */
+const Tooltip = ({
+  children,
+  enterTouchDelay = 500,
+  leaveTouchDelay = 1500,
+  title,
+  wrapperStyle,
+  ...rest
+}: Props) => {
+  const theme = useInternalTheme();
+  const [visible, setVisible] = React.useState(false);
+  const [measurement, setMeasurement] = React.useState({
+    children: {},
+    tooltip: {},
+    measured: false,
+  });
+  const showTooltipTimer = React.useRef<NodeJS.Timeout>();
+  const hideTooltipTimer = React.useRef<NodeJS.Timeout>();
+  const childrenWrapperRef = React.useRef() as React.MutableRefObject<View>;
+
+  React.useEffect(() => {
+    return () => {
+      if (showTooltipTimer.current) clearTimeout(showTooltipTimer.current);
+      if (hideTooltipTimer.current) clearTimeout(hideTooltipTimer.current);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    const subscription = addEventListener(Dimensions, 'change', () =>
+      setVisible(false)
+    );
+
+    return () => subscription.remove();
+  }, []);
+
+  const handleOnLayout = ({ nativeEvent: { layout } }: LayoutChangeEvent) => {
+    childrenWrapperRef.current.measure(
+      (_x, _y, width, height, pageX, pageY) => {
+        setMeasurement({
+          children: { pageX, pageY, height, width },
+          tooltip: { ...layout },
+          measured: true,
+        });
+      }
+    );
+  };
+
+  const handleTouchStart = () => {
+    if (hideTooltipTimer.current) clearTimeout(hideTooltipTimer.current);
+
+    showTooltipTimer.current = setTimeout(
+      () => setVisible(true),
+      enterTouchDelay
+    ) as unknown as NodeJS.Timeout;
+  };
+
+  const handleTouchEnd = () => {
+    if (showTooltipTimer.current) clearTimeout(showTooltipTimer.current);
+
+    hideTooltipTimer.current = setTimeout(() => {
+      setVisible(false);
+      setMeasurement({ children: {}, tooltip: {}, measured: false });
+    }, leaveTouchDelay) as unknown as NodeJS.Timeout;
+  };
+
+  return (
+    <>
+      {visible && (
+        <Portal>
+          <View
+            onLayout={handleOnLayout}
+            style={[
+              styles.tooltip,
+              {
+                ...getTooltipPosition(measurement as Measurement),
+                borderRadius: theme.roundness,
+                ...(measurement.measured ? styles.visible : styles.hidden),
+              },
+            ]}
+            testID="tooltip-container"
+          >
+            <Text
+              accessibilityLiveRegion="polite"
+              numberOfLines={1}
+              selectable={false}
+              variant="labelLarge"
+              style={styles.text}
+            >
+              {title}
+            </Text>
+          </View>
+        </Portal>
+      )}
+      <View
+        onTouchStart={handleTouchStart}
+        onTouchEnd={handleTouchEnd}
+        onTouchCancel={handleTouchEnd}
+        style={wrapperStyle}
+      >
+        {React.cloneElement(children, { ...rest, ref: childrenWrapperRef })}
+      </View>
```

Comment:
🧵 
At first, I tried to use this View to measure the wrapped element as in this change

```suggestion
      <View
        onTouchStart={handleTouchStart}
        onTouchEnd={handleTouchEnd}
        onTouchCancel={handleTouchEnd}
        style={wrapperStyle}
        collapsable={false}
        ref={childrenWrapperRef}
      >
        {React.cloneElement(children, { ...rest })}
      </View>
```

However, I ran into some issues with absolute positioned elements such as a FAB. The only I found to work around this issue was to pass a ref to the wrapped elements as MUI does, see [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Tooltip/Tooltip.js#L724) and more explanation [here](https://mui.com/material-ui/guides/composition/#caveat-with-refs)

Reply:
@lukewalczak I'm wondering if we can have a prop called `forwardRef` for cases when the element is absolutely positioned such as a FAB. Then we wouldn't need to make the components that I changed forwardable. What do you think?

Reply:
I would say to keep your current approach, I think are no benefits of having the prop `forwardRef` and it can be confusing for users.